### ISSUE TITLE
[Bug]: godot can not be rebuilt or upgraded.

### ISSUE THREAD

**Issue by twaik** [termux/termux-packages#20133]
### Problem description

Godot build fails with the following errors.
```
thirdparty/icu4c/common/localematcher.cpp:341:9: error: no viable conversion from 'const XLikelySubtags' to incomplete type 'const LikelySubtags'
        likelySubtags(*XLikelySubtags::getSingleton(errorCode)),
        ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/data/com.termux/files/usr/include/unicode/localematcher.h:138:7: note: forward declaration of 'icu_75_godot::LikelySubtags'
class LikelySubtags;
      ^
thirdparty/icu4c/common/localematcher.cpp:362:29: error: no matching function for call to 'getMaximalLsrOrUnd'
        builderDefaultLSR = getMaximalLsrOrUnd(likelySubtags, *def, errorCode);
                            ^~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:305:5: note: candidate function not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
LSR getMaximalLsrOrUnd(const XLikelySubtags &likelySubtags, const Locale &locale,
    ^
thirdparty/icu4c/common/localematcher.cpp:392:34: error: no matching function for call to 'getMaximalLsrOrUnd'
            LSR &lsr = lsrs[i] = getMaximalLsrOrUnd(likelySubtags, supportedLocale, errorCode);
                                 ^~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:305:5: note: candidate function not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
LSR getMaximalLsrOrUnd(const XLikelySubtags &likelySubtags, const Locale &locale,
    ^
thirdparty/icu4c/common/localematcher.cpp:469:23: error: no matching function for call to 'getMaximalLsrOrUnd'
        LSR suppLSR = getMaximalLsrOrUnd(likelySubtags, *builder.maxDistanceSupported_, errorCode);
                      ^~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:305:5: note: candidate function not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
LSR getMaximalLsrOrUnd(const XLikelySubtags &likelySubtags, const Locale &locale,
    ^
thirdparty/icu4c/common/localematcher.cpp:472:17: error: no matching function for call to 'getMaximalLsrOrUnd'
                getMaximalLsrOrUnd(likelySubtags, *builder.maxDistanceDesired_, errorCode),
                ^~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:305:5: note: candidate function not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
LSR getMaximalLsrOrUnd(const XLikelySubtags &likelySubtags, const Locale &locale,
    ^
thirdparty/icu4c/common/localematcher.cpp:609:9: error: no matching function for call to 'getMaximalLsrOrUnd'
        getMaximalLsrOrUnd(likelySubtags, desiredLocale, errorCode),
        ^~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:305:5: note: candidate function not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
LSR getMaximalLsrOrUnd(const XLikelySubtags &likelySubtags, const Locale &locale,
    ^
thirdparty/icu4c/common/localematcher.cpp:608:13: error: no viable conversion from 'std::optional<int32_t>' (aka 'optional<int>') to 'int32_t' (aka 'int')
    int32_t suppIndex = getBestSuppIndex(
            ^           ~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:620:23: error: no matching constructor for initialization of 'LocaleLsrIterator'
    LocaleLsrIterator lsrIter(likelySubtags, desiredLocales, ULOCMATCH_TEMPORARY_LOCALES);
                      ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:554:5: note: candidate constructor not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
    LocaleLsrIterator(const XLikelySubtags &likelySubtags, Locale::Iterator &locales,
    ^
thirdparty/icu4c/common/localematcher.cpp:552:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 3 were provided
class LocaleLsrIterator {
      ^
thirdparty/icu4c/common/localematcher.cpp:621:13: error: no viable conversion from 'std::optional<int32_t>' (aka 'optional<int>') to 'int32_t' (aka 'int')
    int32_t suppIndex = getBestSuppIndex(lsrIter.next(errorCode), &lsrIter, errorCode);
            ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:638:9: error: no matching function for call to 'getMaximalLsrOrUnd'
        getMaximalLsrOrUnd(likelySubtags, desiredLocale, errorCode),
        ^~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:305:5: note: candidate function not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
LSR getMaximalLsrOrUnd(const XLikelySubtags &likelySubtags, const Locale &locale,
    ^
thirdparty/icu4c/common/localematcher.cpp:637:13: error: no viable conversion from 'std::optional<int32_t>' (aka 'optional<int>') to 'int32_t' (aka 'int')
    int32_t suppIndex = getBestSuppIndex(
            ^           ~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:652:23: error: no matching constructor for initialization of 'LocaleLsrIterator'
    LocaleLsrIterator lsrIter(likelySubtags, desiredLocales, ULOCMATCH_TEMPORARY_LOCALES);
                      ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[Initial build] thirdparty/icu4c/common/localematcher.cpp:554:5: note: candidate constructor not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
    LocaleLsrIterator(const XLikelySubtags &likelySubtags, Locale::Iterator &locales,
    ^
[Initial build] thirdparty/icu4c/common/localematcher.cpp:552:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 3 were provided
class LocaleLsrIterator {
      ^
thirdparty/icu4c/common/localematcher.cpp:653:13: error: no viable conversion from 'std::optional<int32_t>' (aka 'optional<int>') to 'int32_t' (aka 'int')
    int32_t suppIndex = getBestSuppIndex(lsrIter.next(errorCode), &lsrIter, errorCode);
            ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:662:24: error: return type of out-of-line definition of 'icu_75_godot::LocaleMatcher::getBestSuppIndex' differs from that in the declaration
int32_t LocaleMatcher::getBestSuppIndex(LSR desiredLSR, LocaleLsrIterator *remainingIter,
~~~~~~~                ^
[Initial build] /data/data/com.termux/files/usr/include/unicode/localematcher.h:683:28: note: previous declaration is here
    std::optional<int32_t> getBestSuppIndex(LSR desiredLSR, LocaleLsrIterator *remainingIter, UErrorCode &errorCode) const;
    ~~~~~~~~~~~~~~~~~~~~~~ ^
thirdparty/icu4c/common/localematcher.cpp:710:19: error: no matching function for call to 'getMaximalLsrOrUnd'
    LSR suppLSR = getMaximalLsrOrUnd(likelySubtags, supported, errorCode);
                  ^~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:305:5: note: candidate function not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
LSR getMaximalLsrOrUnd(const XLikelySubtags &likelySubtags, const Locale &locale,
    ^
thirdparty/icu4c/common/localematcher.cpp:714:13: error: no matching function for call to 'getMaximalLsrOrUnd'
            getMaximalLsrOrUnd(likelySubtags, desired, errorCode),
            ^~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:305:5: note: candidate function not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
LSR getMaximalLsrOrUnd(const XLikelySubtags &likelySubtags, const Locale &locale,
    ^
thirdparty/icu4c/common/localematcher.cpp:722:19: error: no matching function for call to 'getMaximalLsrOrUnd'
    LSR suppLSR = getMaximalLsrOrUnd(likelySubtags, supported, errorCode);
                  ^~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:305:5:/home/twaik/.termux-build/_cache/android-r26b-api-24-v3/bin/aarch64-linux-android-clang++ -o thirdparty/icu4c/common/locutil.linuxbsd.editor.arm64.llvm.o -c -std=gnu++17 -I/data/data/com.termux/files/usr/include -fstack-protector-strong -Oz -fno-exceptions -DU_STATIC_IMPLEMENTATION -DU_COMMON_IMPLEMENTATION -DUCONFIG_NO_COLLATION -DUCONFIG_NO_CONVERSION -DUCONFIG_NO_FORMATTING -DUCONFIG_NO_SERVICE -DUCONFIG_NO_IDNA -DUCONFIG_NO_FILE_IO -DUCONFIG_NO_TRANSLITERATION -DUCONFIG_NO_REGULAR_EXPRESSIONS -DPKGDATA_MODE=static -DU_ENABLE_DYLOAD=0 -DU_HAVE_LIB_SUFFIX=1 -DU_LIB_SUFFIX_C_NAME=_godot -DICU_DATA_NAME=icudt73l.dat -pipe -O2 -Wall -Wshadow-field-in-constructor -Wshadow-uncaptured-local -Wno-ordered-compare-function-pointers -w -isystem thirdparty/glad -DTOOLS_ENABLED -DDEBUG_ENABLED -DNDEBUG -DNO_EDITOR_SPLASH -DSOWRAP_ENABLED -DTOUCH_ENABLED -DFONTCONFIG_ENABLED -DPULSEAUDIO_ENABLED -D_REENTRANT -DDBUS_ENABLED -DSPEECHD_ENABLED -DXKB_ENABLED -DJOYDEV_ENABLED -DLINUXBSD_ENABLED -DUNIX_ENABLED -D_FILE_OFFSET_BITS=64 -DX11_ENABLED -DVULKAN_ENABLED -DGLES3_ENABLED -DMINIZIP_ENABLED -DBROTLI_ENABLED -DCLIPPER2_ENABLED -DUSE_VOLK -DVK_USE_PLATFORM_XLIB_KHR -DGLAD_ENABLED -DEGL_ENABLED -Ithirdparty/icu4c/common -Ithirdparty/icu4c/i18n -Ithirdparty/libpng -Ithirdparty/volk -Ithirdparty/vulkan -Ithirdparty/vulkan/include -Ithirdparty/zlib -Ithirdparty/clipper2/include -Ithirdparty/brotli/include -Ithirdparty/linuxbsd_headers -Iplatform/linuxbsd -I/usr/include/miniupnpc -I. -I/data/data/com.termux/files/usr/include -I/data/data/com.termux/files/usr/include/webp thirdparty/icu4c/common/locutil.cpp
 note: candidate function not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
LSR getMaximalLsrOrUnd(const XLikelySubtags &likelySubtags, const Locale &locale,
    ^
thirdparty/icu4c/common/localematcher.cpp:726:13: error: no matching function for call to 'getMaximalLsrOrUnd'
            getMaximalLsrOrUnd(likelySubtags, desired, errorCode),
            ^~~~~~~~~~~~~~~~~~
thirdparty/icu4c/common/localematcher.cpp:305:5: note: candidate function not viable: cannot convert argument of incomplete type 'const LikelySubtags' to 'const XLikelySubtags' for 1st argument
LSR getMaximalLsrOrUnd(const XLikelySubtags &likelySubtags, const Locale &locale,
    ^
```

### What steps will reproduce the bug?

You can update version to `4.2.2` with `990b7b716656122364b1672508c516c898497c50216d7c00c60eeaf507685c0e` or not. The problem is reproducible with both `4.2.1` and `4.2.2` versions.
`./build-package -I -f godot`

### What is the expected behavior?

It should be built.

### System information

```shell
Irrelevant, built is performed on host, not on device.
```

**Comment by twaik** [comment]
If you will upgrade version to `4.2.2` please, do not forget to upgrade `libenet` to `1.3.18` with sha `2a8a0c5360d68bb4fcd11f2e4c47c69976e8d2c85b109dd7d60b1181a4f85d36` too.

**Comment by licy183** [comment]
Emmm... I can't reproduce it with `./build-package -i godot`.

**Comment by twaik** [comment]
Probably a problem of a host environment.

**Comment by twaik** [comment]
Fixed by 800bdb957afadc1491437ab7448dfc1111cca3aa
